### PR TITLE
chore: store tokio::runtime::Handle in ethers/alloyDB

### PR DESCRIPTION
**Pull request by chirag-bgh**
ref #1534

### CODE REVIEW

**Review by DaniPopes — CHANGES_REQUESTED** [review]
can you also return None if the runtime flavor is CurrentThread, and can you document it briefly in the fn new doc comments saying that it requires a tokio runtime?

otherwise looks good

File: crates/revm/src/db/ethersdb.rs [hunk 1]
```diff
@@ -11,55 +11,45 @@ use crate::{Database, DatabaseRef};
...
             block_number,
+            handle,
         })
     }
 
-    /// internal utility function to call tokio feature and wait for output
+    /// Internal utility function to call tokio feature and wait for output
     #[inline]
-    fn block_on<F>(f: F) -> F::Output
+    fn block_on<F>(f: F, handle: &Handle) -> F::Output
```
Comment (mattsse) [inline crates/revm/src/db/ethersdb.rs:47]: this function can now accept &self

**Review by mattsse — DISMISSED** [review]

**Review by DaniPopes — APPROVED** [review]

**Review by rakita — APPROVED** [review]

**Comment by wtdcode** [comment]
@rakita  This change is not appropriate. If you wish to enforce ethersdb/alloydb to always play with async world, you should mark `fn new` as `async fn new` even not using any `await` in the function. This removes the check and also presents the correct signature.

@DaniPopes I disagree with not allowing ethersdb/alloydb to create current thread runtime. This makes them totally incompatible with sync world and creating a current thread runtime is rather cheap.

**Comment by wtdcode** [comment]
@DaniPopes Also, why would you disallow the current thread flavor? I can't understand.

**Pull request merged**